### ISSUE TITLE
Enable TypeScript `strict` mode

### ISSUE THREAD

**Issue by mondash** [shadcn/taxonomy#49]
TypeScript strict mode enables some extremely useful static type analysis and prevents some easy to stumble into pitfalls. Unless there's a specific compatibility reason why it needs to be disabled, enabling it would be beneficial to the type-safety and runtime stability of this app

I originally realized it wasn't enabled after I wasn't able to get proper type inference using `zod`. Specifically, the discriminated union return type of `parseSafe` isn't able to be narrowed to one case unless strict mode is enabled.

There _are_ some unhandled type errors when enabling strict mode. Most of these are either accessing a property of a nullish object or assigning a nullish value to a non-nullable target. I'd be willing to do a PR to enable strict mode and resolve these type errors but figured I'd raise an issue first.

**Comment by shadcn** [comment]
Agreed. 💯 I'm all for strict mode. This is was on when I started developing the app. 

The reason I had to turn it off was Next (the early releases) was throwing errors and wouldn't build.

I'm open to turning this on again and giving it another go.

Happy to review a PR if you're up for it.

Thank you.

**Comment by AliMejbar** [comment]
Any update on this ?

**Comment by shadcn** [comment]
I'll get back to this. Hopefully next week.

**Comment by mondash** [comment]
Appreciate it! I fell behind on this one, and there were also some decisions to make about control flow and nullability that I didn't think I should be the one making.

**Comment by Dannymx** [comment]
As I was checking this repo for examples, I decided to do a quick `strict: true` and fix the errors the rule triggered. [PR here](https://github.com/shadcn/taxonomy/pull/71).

**Comment by Falven** [comment]
Not sure what triggered it, but I am getting a ton of errors, mostly around `Type error: Object is possibly 'undefined'.`. Not sure if it's because of my Typescript version `"typescript": "^5.7.2",` or if my `tsconfig.json` is not correct for `shadcn/ui`. I am using the [Turborepo example](https://github.com/vercel/turborepo/blob/main/examples/basic/apps/docs/tsconfig.json) configuration. I am fixing them all one by one, but they are seemingly endless.
